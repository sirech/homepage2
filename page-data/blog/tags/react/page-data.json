{
    "componentChunkName": "component---src-templates-blog-index-js",
    "path": "/blog/tags/react",
    "result": {"pageContext":{"siteMetadata":{"title":"Mario Fernandez","description":"This is the blog from Mario Fernandez, a Software Developer working for ThoughtWorks","author":"Mario Fernandez","url":"https://hceris.com","twitter":"@sirech"},"group":[{"frontmatter":{"layout":"post","title":"Applications Instead of Libraries","path":"/applications-instead-of-libraries/","categories":["Architecture","React","Micro frontends","JavaScript"],"date":"2021/11/20","draft":false,"description":"Using module federation to implement micro frontends"}},{"frontmatter":{"layout":"post","title":"From Redux to the Context API: A Practical Migration Guide","path":"/redux-to-context-practical-migration-guide/","categories":["React","Redux","Context API","Architecture","State Management"],"date":"2021/06/14","draft":false,"description":"Let's migrate a React application built with Redux to the Context API and see how it looks like"}},{"frontmatter":{"layout":"post","title":"Starting you new React application with Redux? Consider a simpler approach using the Context API","path":"/starting-a-new-app-with-redux-consider-context-api-first/","categories":["React","Redux","Context API","Architecture"],"date":"2021/05/11","draft":false,"description":"Traditionally, starting a new React application means adding Redux for state management. Let's talk about the Context API as an alternative"}},{"frontmatter":{"layout":"post","title":"Authorization Code Flow with PKCE (OAuth) in a React application","path":"/oauth-authorization-code-flow-pkce-for-react/","categories":["Auth0","OAuth","React"],"date":"2021/03/29","draft":false,"description":"SPAs implementing OAuth should pick Authorization Code Flow with PKCE for maximum security. Let's implement it in React with help of Auth0"}},{"frontmatter":{"layout":"post","title":"A directory structure for React projects","path":"/a-directory-structure-for-react-projects/","categories":["React","TypeScript","Code Organization"],"date":"2020/07/19","draft":false,"description":"How do you structure your components in a React application? I like to group mine around features close to the domain enforced by convention"}},{"frontmatter":{"layout":"post","title":"Fail a test in Jest if an unexpected network request happens","path":"/jest-fail-test-if-unexpected-network-request-happens/","categories":["React","Jest","Testing","react-testing-library","create-react-app","axios"],"date":"2019/12/08","draft":false,"description":"Automatically failing unit tests with Jest in React when network requests are made will make your tests more reliable and easier to maintain"}},{"frontmatter":{"layout":"post","title":"Custom Components in Formik","path":"/custom-components-in-formik/","categories":["React","Formik","react-hooks"],"date":"2019/08/15","draft":false,"description":"Formik is the new cool kid on the block for forms in react. Here is how to build custom input elements on top"}},{"frontmatter":{"layout":"post","title":"Migrating to Gatsby 2","path":"/migrating-to-gatsby2/","categories":["React","Gatsby"],"date":"2018/10/03","draft":false,"description":"Gatsby 2 is out! How did the migration go?"}},{"frontmatter":{"layout":"post","title":"Angular from the perspective of a React fan - Part 3","path":"/angular-from-react-part3/","categories":["JavaScript","React","Angular","TypeScript","angular-cli","redux"],"date":"2018/07/07","draft":false,"description":"Third part of my series on Angular, coming from a React background. This article focus on state management"}},{"frontmatter":{"layout":"post","title":"Angular from the perspective of a React fan - Part 2","path":"/angular-from-react-part2/","categories":["JavaScript","React","Angular","TypeScript","angular-cli"],"date":"2018/04/29","draft":false,"description":"Second part of my series on Angular, coming from a React background. This article focus on testing"}},{"frontmatter":{"layout":"post","title":"Angular from the perspective of a React fan - Part 1","path":"/angular-from-react-part1/","categories":["JavaScript","React","Angular","TypeScript","angular-cli"],"date":"2018/03/25","draft":false,"description":"First part of my series on Angular, coming from a React background. This article is about bootstrapping and TypeScript"}},{"frontmatter":{"layout":"post","title":"Migrating to Gatsby","path":"/migrating-to-gatsby/","categories":["React","Gatsby"],"date":"2018/03/18","draft":false,"description":"It all starts here. After using Create React App for a while, I decided to switch to Gatsby for my personal website"}}],"pathPrefix":"blog/tags/react","first":true,"last":true,"index":1,"pageCount":1,"additionalContext":{"siteMetadata":{"title":"Mario Fernandez","description":"This is the blog from Mario Fernandez, a Software Developer working for ThoughtWorks","author":"Mario Fernandez","url":"https://hceris.com","twitter":"@sirech"}}}},
    "staticQueryHashes": []}