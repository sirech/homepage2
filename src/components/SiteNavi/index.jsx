import React from 'react'
import Link from 'gatsby-link'

const links = [['/', 'About'], ['/talks/', 'Talks'], ['/blog/', 'Blog']]

class SiteNavi extends React.Component {
  link(url, name) {
    return (
      <li
        key={name}
        className={
          this.props.location.pathname === url ? 'nav-item active' : 'nav-item'
        }
      >
        <Link to={url} className="nav-link">
          {name}
        </Link>
      </li>
    )
  }

  render() {
    const { title } = this.props
    return (
      <nav className="navbar navbar-expand navbar-dark flex-column flex-md-row bg-dark">
        <div className="container">
          <Link className="text-center" to="/">
            <h1 className="navbar-brand mb-0">{title}</h1>
          </Link>
          <div className="navbar-nav-scroll">
            <ul className="navbar-nav bd-navbar-nav flex-row">
              {links.map(([url, name]) => this.link(url, name))}
            </ul>
          </div>
          <div className="navbar-nav flex-row ml-md-auto d-none d-md-flex" />
        </div>
      </nav>
    )
  }
}

export default SiteNavi
