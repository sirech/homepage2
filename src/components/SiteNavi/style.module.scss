@import '../../layouts/colors';

.underline {
  position: relative;

  &::before {
    background-color: $white;
    bottom: 0.375rem;
    content: '';
    height: 1px;
    left: 0.5rem;
    opacity: 0;
    position: absolute;
    right: 100%;
    transition: all 0.5s ease-in-out;
  }

  &:hover {
    &::before {
      opacity: 0.5;
      right: 0.5rem;
    }
  }
}

.icon {
  transition: all 0.5s ease;

  &:hover {
    transform: scale(1.5);
  }
}
