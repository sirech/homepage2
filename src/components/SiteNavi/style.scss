.navbar-icons li {
  transition: all 0.5s ease;

  &:hover {
    transform: scale(1.5);
  }
}
