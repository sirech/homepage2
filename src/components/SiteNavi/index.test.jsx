import React from 'react'
import { render } from '@testing-library/react'
import SiteNavi from './index'

describe('components', () => {
  describe('SiteNavi', () => {
    it('renders correctly', () => {
      const location = { pathname: '/' }
      const { asFragment } = render(
        <SiteNavi title="title" location={location} />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
