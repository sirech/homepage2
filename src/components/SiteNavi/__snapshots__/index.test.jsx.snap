// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components SiteNavi renders correctly 1`] = `
<nav
  className="navbar navbar-expand navbar-dark flex-column flex-md-row bg-dark"
>
  <a
    className="text-center"
    href="/"
  >
    <h1
      className="navbar-brand mb-0 mr-sm-0 mr-md-2"
    >
      title
    </h1>
  </a>
  <div
    className="navbar-nav-scroll"
  >
    <ul
      className="navbar-nav bd-navbar-nav flex-row"
    >
      <li
        className="nav-item active"
      >
        <a
          className="underline nav-link"
          href="/"
        >
          About
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="underline nav-link"
          href="/talks/"
        >
          Talks
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="underline nav-link"
          href="/blog/"
        >
          Blog
        </a>
      </li>
    </ul>
  </div>
  <ul
    className="navbar-icons navbar-nav flex-row ml-md-auto d-md-flex"
  >
    <li
      className="icon nav-item"
    >
      <a
        aria-label="RSS Feed"
        className="nav-link"
        href="/rss.xml"
      >
        <i
          className="fa fa-rss fa-lg"
        />
      </a>
    </li>
    <li
      className="icon nav-item"
    >
      <a
        aria-label="StackOverflow Profile"
        className="nav-link"
        href="https://github.com/sirech"
        rel="noopener noreferrer"
        target="_blank"
      >
        <i
          className="fa fa-github-square fa-lg"
        />
      </a>
    </li>
    <li
      className="icon nav-item"
    >
      <a
        aria-label="StackOverflow Profile"
        className="nav-link"
        href="https://stackoverflow.com/users/3785/mario-f"
        rel="noopener noreferrer"
        target="_blank"
      >
        <i
          className="fa fa-stack-overflow fa-lg"
        />
      </a>
    </li>
  </ul>
</nav>
`;
